apps/form-gateway/src/gateway/router.ts

```diff
@@ -62,10 +101,29 @@ async function getFile(
   return fileId;
 }
 
-export function findFile(fileApiUrl: URL, tokenProvider: TokenProvider): RequestHandler {
+export function canAccessFile(formResult: SimpleFormResponse, roles: string[], criteria: string) {
+  if (criteria === '{}') return false;
+  if (formResult === null) return false;
+
+  return (
+    formResult?.status === FormStatus.Submitted &&
+    formResult?.submitted !== null &&
+    //Need to check the form gateway roles and form service roles, depending where the request was made.
+    //Whether it was from the form app or directly through the form gateway the roles might be different.
+    roles.find((role) => role.includes(FormServiceRoles.Applicant) || role.includes(ServiceRoles.Applicant))
+  );
+}
+
+export function findFile(fileApiUrl: URL, formApiUrl: URL, tokenProvider: TokenProvider): RequestHandler {
   return async (req, res, next) => {
```

Comment:
Setting aside the user permissions aspect of this, the two endpoints are just too broad in functionality. The requirement for the gateway is to allow an applicant to download the PDF generated for their form submission. So for example, the endpoint should only take the necessarily parameters (i.e. form ID and submission ID if the form creates submission records). The path does not have to match the file API endpoint path, so for example it could be `/download/:formId/:submissionId` or `/download?formId=:formId&submissionId=:submissionI`. There no need to support a criteria object as a query parameter, and no need to pass the user provided value directly through to the file API.